internal/provisioner/objects/service/service.go

```diff
@@ -250,6 +250,10 @@ func DesiredEnvoyService(contour *model.Contour) *corev1.Service {
 		}
 	}
 
+	if contour.Spec.NetworkPublishing.Envoy.IPFamilyPolicy != "" {
+		svc.Spec.IPFamilyPolicy = &contour.Spec.NetworkPublishing.Envoy.IPFamilyPolicy
```

Comment:
i *think* this could technically cause a memory leak since we're keeping a reference to a struct field so the containing struct may not be GCed, could instead use `ref.To` helper which would make a copy of the value of the field only and return a reference to the copy

there may be other places in the codebase we should fix this as well